### PR TITLE
Rename RawJSON to Raw in runtime.Unknown and add ContentType & Conten…

### PR DESCRIPTION
**Pull request by wojtek-t**
…tEncoding.

### CODE REVIEW

File: pkg/runtime/types.go [hunk 1]
```diff
@@ -96,10 +100,16 @@ type RawExtension struct {
 // +protobuf=true
 type Unknown struct {
 	TypeMeta `json:",inline"`
-	// RawJSON will hold the complete JSON of the object which couldn't be matched
+	// Raw will hold the complete serialized object which couldn't be matched
 	// with a registered type. Most likely, nothing should be done with this
 	// except for passing it through the system.
-	RawJSON []byte
+	Raw []byte
```
Comment (wojtek-t) [inline pkg/runtime/types.go]: @smarterclayton - I guess we should also change RawExtension. Can I do it in a separate PR?

Comment (smarterclayton) [inline pkg/runtime/types.go]: Yes

File: pkg/runtime/embedded.go [hunk 2]
```diff
@@ -69,17 +69,22 @@ func (re *Unknown) UnmarshalJSON(in []byte) error {
...
+	re.ContentEncoding = ""
+	re.ContentType = ContentTypeJSON
 	return nil
 }
 
 // Marshal may get called on pointers or values, so implement MarshalJSON on value.
 // http://stackoverflow.com/questions/21390979/custom-marshaljson-never-gets-called-in-go
 func (re Unknown) MarshalJSON() ([]byte, error) {
-	if re.RawJSON == nil {
+	if re.ContentType != ContentTypeJSON {
```
Comment (smarterclayton) [inline pkg/runtime/embedded.go]: It should be allowed for content type to be empty, at which point it would default to JSON (as per the proposal).  I think that logic has to be here (as well as other places).

File: pkg/runtime/unstructured.go [hunk 3]
```diff
@@ -67,7 +67,7 @@ func (unstructuredJSONScheme) EncodeToStream(obj Object, w io.Writer, overrides
 		}
 		return json.NewEncoder(w).Encode(eList)
 	case *Unknown:
-		_, err := w.Write(t.RawJSON)
+		_, err := w.Write(t.Raw)
```
Comment (smarterclayton) [inline pkg/runtime/unstructured.go]: Add a TODO here that unstructured needs to deal with Content-Type.

File: pkg/runtime/embedded.go (hunk 2 above)
Comment (wojtek-t) [inline pkg/runtime/embedded.go]: Done.

File: pkg/runtime/unstructured.go (hunk 3 above)
Comment (wojtek-t) [inline pkg/runtime/unstructured.go]: Done.

**Comment by wojtek-t** [comment]
PTAL

File: pkg/runtime/embedded.go [hunk 4]
```diff
@@ -69,17 +69,23 @@ func (re *Unknown) UnmarshalJSON(in []byte) error {
...
+	re.ContentEncoding = ""
+	re.ContentType = ContentTypeJSON
 	return nil
 }
 
 // Marshal may get called on pointers or values, so implement MarshalJSON on value.
 // http://stackoverflow.com/questions/21390979/custom-marshaljson-never-gets-called-in-go
 func (re Unknown) MarshalJSON() ([]byte, error) {
-	if re.RawJSON == nil {
+	// If ContentType is unset, we assume this is JSON.
```
Comment (smarterclayton) [inline pkg/runtime/embedded.go]: I'll deal with this in the proposal, but I think internally (In code) empty contentType must mean JSON.  In serialized protobuf objects (from the outside) it means "protobuf encoding".  The code that creates Unknowns will have to get stricter - it may be that when we add the protobuf serializer (and unknown handling) we have to explicitly remove this check, and find all the places in the code where it breaks.  I'll do that as part of the serializer PR.

**Comment by smarterclayton** [comment]
LGTM
